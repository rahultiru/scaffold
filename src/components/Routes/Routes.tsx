import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

// import { Home } from './loadableComponents';

const Routes: React.SFC = () => {
  return (
    <Router>
      <Switch>
        <Route exact={true} path="/" component={() => <div />} />
        <Route exact={true} path="/portfolio" component={() => <div className="portfolio" />} />
      </Switch>
    </Router>
  );
};

export default Routes;
