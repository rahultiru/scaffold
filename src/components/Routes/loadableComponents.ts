// import Spinner from 'components/PageSpinner';
// import lazyComponent, { LazyLoader } from 'yammer-web-components/dist/lazy';

// const homeLoader = () => import('components/PageHome');
// const portfolioLoader = () => import('components/PagePortfolio');

// const lazyComponentWithSpinner = (loader: LazyLoader) => lazyComponent({ loader, loading: Spinner });

// export const Home = lazyComponentWithSpinner(homeLoader);
// export const Portfolio = lazyComponentWithSpinner(portfolioLoader);