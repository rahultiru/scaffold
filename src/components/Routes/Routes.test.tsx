jest.mock('./loadableComponents', () => ({
  Home: jest.fn(),
  Inbox: jest.fn(),
  Notifications: jest.fn(),
}));

import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Routes from './Routes';

describe('Routes', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Routes />);
  });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });
});
