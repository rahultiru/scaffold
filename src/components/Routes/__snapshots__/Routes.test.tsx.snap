// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/:networkPermalink/home"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/:networkPermalink/inbox"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/:networkPermalink/notifications"
  />
  <Route
    exact={true}
    path="/:networkPermalink/groups/:groupId"
  />
  <Route
    exact={true}
    path="/:networkPermalink/threads/:threadId"
  />
  <Route
    exact={true}
    path="/:networkPermalink/users/:userId"
  />
  <Connect(Redirect) />
</Switch>
`;
